tests/unit/lms/validation/_canvas_test.py

```diff
@@ -11,25 +11,117 @@
+            "updated_at": "2019-05-08T15:22:31Z",
+            "upload_status": "success",
+            "url": "TEST_URL_1",
+            "uuid": "TEST_UUID_1",
+            "workflow_state": "processing",
+        },
+        {
+            "content-type": "application/pdf",
+            "created_at": "2018-10-25T15:04:08Z",
+            "display_name": "TEST FILE 2",
+            "filename": "TEST_FILE_2.pdf",
+            "folder_id": 17,
+            "hidden": False,
+            "hidden_for_user": False,
+            "id": 181,
+            "lock_at": None,
+            "locked": False,
+            "locked_for_user": False,
+            "media_entry_id": None,
+            "mime_class": "pdf",
+            "modified_at": "2018-10-25T15:04:08Z",
+            "size": 1407214,
+            "thumbnail_url": None,
+            "unlock_at": None,
+            "updated_at": "2019-02-14T00:33:01Z",
+            "upload_status": "success",
+            "url": "TEST_URL_2",
+            "uuid": "TEST_UUID_2",
+            "workflow_state": "processing",
+        },
+        {
+            "content-type": "application/pdf",
+            "created_at": "2017-09-08T11:05:03Z",
+            "display_name": "TEST FILE 3",
+            "filename": "TEST_FILE_3.pdf",
+            "folder_id": 17,
+            "hidden": False,
+            "hidden_for_user": False,
+            "id": 97,
+            "lock_at": None,
+            "locked": False,
+            "locked_for_user": False,
+            "media_entry_id": None,
+            "mime_class": "pdf",
+            "modified_at": "2017-09-08T11:05:03Z",
+            "size": 265615,
+            "thumbnail_url": None,
+            "unlock_at": None,
+            "updated_at": "2018-10-19T17:16:50Z",
+            "upload_status": "success",
+            "url": "TEST_URL_3",
+            "uuid": "TEST_UUID_3",
+            "workflow_state": "processing",
+        },
+    ]
+    return CanvasListFilesResponseSchema(response_)
+
+
+def canvas_public_url_response_schema():
+    """Return a CanvasPublicURLResponseSchema."""
+    response_ = response()
+    response_.json.return_value = {
+        "public_url": "https://example-bucket.s3.amazonaws.com/example-namespace/attachments/1/example-filename?AWSAccessKeyId=example-key&Expires=1400000000&Signature=example-signature"
+    }
+    return CanvasPublicURLResponseSchema(response_)
+
+
+class TestCommon:
     def test_it_raises_ValidationError_if_the_response_json_has_the_wrong_format(
-        self, response, schema
+        self, schema
     ):
         # The decoded JSON value is a string rather than a list or object.
-        response.json.return_value = "wrong_format"
-        schema.context["response"] = response
+        schema.context["response"].json.return_value = "wrong_format"
 
         with pytest.raises(ValidationError) as exc_info:
             schema.parse()
 
         assert exc_info.value.messages == {"_schema": ["Invalid input type."]}
```

Comment:
May be, although if `CanvasListFilesResponseSchema` or `CanvasPublicURLResponseSchema` no longer inherited from `RequestsResponseSchema` this test would fail. If the code used composition rather than inheritance this could be a "it raises if `RequestsResponseSchema` raises" test. I'm gonna draw the line before second-guessing this old test for now anyway, in this PR just detangling the test classes